perform.pl

```diff
@@ -90,27 +106,57 @@ sub good {
     $tr->sync();
 }
 
-my $local_addr = $ENV{LOCAL_ADDR}
-    or die "Environemnt LOCAL_ADDR not set";
-my $remote_addr = $ENV{REMOTE_ADDR}
-    or die "Environemnt REMOTE_ADDR not set";
 my $remote_ssh = $ENV{REMOTE_SSH}
     or die "Environemnt REMOTE_SSH not set";
 
+if ($testmode{iperftcp4} || $testmode{iperfudp4} ||
+	$testmode{tcpbench4} || $testmode{udpbench4}) {
+    my $local_addr = $ENV{LOCAL_ADDR}
```

Comment:
The variable is scoped to the block.  How can it be used after the if block?  Does perl -c compile.
Just set the address and address6 environment unconditionally.

Reply:
How about
```perl
my $remote_ssh = $ENV{REMOTE_SSH}
    or die "Environemnt REMOTE_SSH not set";
my $local_addr = $ENV{LOCAL_ADDR};
my $remote_addr = $ENV{REMOTE_ADDR};
my $local_addr6 = $ENV{LOCAL_ADDR6};
my $remote_addr6 = $ENV{REMOTE_ADDR6};

die "Environemnt LOCAL_ADDR and REMOTE_ADDR required"
    if ((!defined $local_addr || !defined $remote_addr) &&
    ($testmode{iperftcp4} || $testmode{iperfudp4} ||
    $testmode{tcpbench4} || $testmode{udpbench4})); 
die "Environemnt LOCAL_ADDR6 and REMOTE_ADDR6 required"
    if ((!defined $local_addr || !defined $remote_addr) &&
    ($testmode{iperftcp6} || $testmode{iperfudp6} ||
    $testmode{tcpbench6} || $testmode{udpbench6}));
```

Reply:
Do not make it too complicated.  Just die if any variable is not set.  I configured them on all perform test machines.